tests/suites/test_suite_nist_kw.function

```diff
@@ -1,159 +1,159 @@
+                                    &output_len));
 
     TEST_ASSERT(output_len == sizeof(ciphertext2));
 
-    TEST_ASSERT(mbedtls_nist_kw_setkey(&ctx2,
-                                       MBEDTLS_CIPHER_ID_AES,
-                                       key, sizeof(key) * 8,
-                                       0) == 0);
-
-    TEST_ASSERT(mbedtls_nist_kw_unwrap(&ctx2, MBEDTLS_KW_MODE_KW,
-                                       ciphertext1, sizeof(ciphertext1),
-                                       plaintext, &output_len,
-                                       sizeof(plaintext)) == 0);
+    PSA_ASSERT(mbedtls_nist_kw_unwrap(key_id_2, MBEDTLS_KW_MODE_KW,
+                                      ciphertext1, sizeof(ciphertext1),
+                                      plaintext, sizeof(plaintext),
+                                      &output_len));
 
     TEST_ASSERT(output_len == sizeof(plaintext));
 
     for (i = 0; i < sizeof(plaintext); i++) {
         TEST_ASSERT(plaintext[i] == 0);
     }
 
-    TEST_ASSERT(mbedtls_nist_kw_unwrap(&ctx2, MBEDTLS_KW_MODE_KWP,
-                                       ciphertext2, sizeof(ciphertext2),
-                                       plaintext, &output_len,
-                                       sizeof(plaintext)) == 0);
+    PSA_ASSERT(mbedtls_nist_kw_unwrap(key_id_2, MBEDTLS_KW_MODE_KWP,
+                                      ciphertext2, sizeof(ciphertext2),
+                                      plaintext, sizeof(plaintext),
+                                      &output_len));
 
     TEST_ASSERT(output_len == sizeof(plaintext));
 
     for (i = 0; i < sizeof(plaintext); i++) {
         TEST_ASSERT(plaintext[i] == 0);
     }
-
 exit:
-    mbedtls_nist_kw_free(&ctx1);
-    mbedtls_nist_kw_free(&ctx2);
+    psa_destroy_key(key_id_1);
+    psa_destroy_key(key_id_2);
+    PSA_DONE();
 }
 /* END_CASE */
 
 /* BEGIN_CASE */
-void mbedtls_nist_kw_setkey(int cipher_id, int key_size,
-                            int is_wrap, int result)
-{
-    mbedtls_nist_kw_context ctx;
-    unsigned char key[32];
-    int ret;
-
-    mbedtls_nist_kw_init(&ctx);
-
-    memset(key, 0x2A, sizeof(key));
-    TEST_ASSERT((unsigned) key_size <= 8 * sizeof(key));
-
-    ret = mbedtls_nist_kw_setkey(&ctx, cipher_id, key, key_size, is_wrap);
-    TEST_ASSERT(ret == result);
-
-exit:
-    mbedtls_nist_kw_free(&ctx);
-}
-/* END_CASE */
-
-/* BEGIN_CASE depends_on:MBEDTLS_AES_C */
 void nist_kw_plaintext_lengths(int in_len, int out_len, int mode, int res)
 {
-    mbedtls_nist_kw_context ctx;
-    unsigned char key[16];
+    mbedtls_svc_key_id_t key_id = MBEDTLS_SVC_KEY_ID_INIT;
+    data_t key;
     unsigned char *plaintext = NULL;
     unsigned char *ciphertext = NULL;
     size_t output_len = out_len;
+    uint8_t key_array[16];
+    key.len = 16;
+    key.x = key_array;
 
-    mbedtls_nist_kw_init(&ctx);
-
-    memset(key, 0, sizeof(key));
+    memset(key_array, 0, sizeof(key_array));
```

Comment:
I prefer to initialize things as early as possible, thus
```suggestion
    uint8_t key_array[16] = { 0 };
    data_t key = { .len = sizeof(key_array), .x = key_array };
```
rather than relying on assignments and memset further below. This way there's no risk of forgetting to initialize on some code paths (e.g. in case of an early error).